cmd/main.go

```diff
@@ -15,23 +16,49 @@ import (
 	"github.com/paulinshek/cryptic-colab/internal/pkg/dataaccess"
 )
 
-func homeLink(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "Welcome home!")
-}
-
 func init() {
+	log.Print("Initialising")
 	if err := godotenv.Load(); err != nil {
 		log.Print("No .env file found")
 	}
 }
 
+type webHandler struct {
+	staticPath string
+	indexPath  string
+}
+
+func (h webHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	path, err := filepath.Abs(r.URL.Path)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	path = filepath.Join(h.staticPath, path)
+
+	_, err = os.Stat(path)
+	if os.IsNotExist(err) {
+		http.ServeFile(w, r, filepath.Join(h.staticPath, h.indexPath))
+		return
+	} else if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	http.FileServer(http.Dir(h.staticPath)).ServeHTTP(w, r)
+}
```

Comment:
this is needed for serving the static files for the front end. i'll move to a different file